devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/Deploy.java

```diff
@@ -71,55 +85,120 @@ public void setImageBuilder(String imageBuilder) {
-                .map(d -> d.getName())
-                .collect(Collectors.toList());
-
-        if (!projectDependencies.contains(requiredDeployerExtension)) {
-            abort("Task: {} requires extensions: {}\n" +
-                    "To add the extensions to the project you can run the following command:\n" +
-                    "\tgradle addExtension --extensions={}",
-                    getName(), requiredDeployerExtension,
-                    requiredDeployerExtension);
-        } else if (!requiredContainerImageExtension.map(b -> projectDependencies.stream().anyMatch(d -> d.equals(b)))
-                .orElse(true)) {
-            abort("Task: {} using: {} is explicitly configured with missing container image builder extension: {}. Aborting.",
-                    getName(), deployer.name(), requiredContainerImageExtension.get());
-        } else if (imageBuild && deployer.getRequiresOneOf().length > 0) {
-            List<String> unsatisfied = Arrays.stream(deployer.requiresOneOf)
-                    .filter(r -> !projectDependencies.stream().anyMatch(d -> d.equals(r)))
-                    .collect(Collectors.toList());
-            if (unsatisfied.size() == deployer.getRequiresOneOf().length) {
-                abort("Task: {} using: {} requires one of the following container image extensions: {}. Aborting.", getName(),
-                        deployer.name(), unsatisfied.stream().collect(Collectors.joining(", ", "[", "]")));
+        ApplicationModel appModel = resolveAppModelForBuild();
+        Map<String, String> configMap = extension().buildEffectiveConfiguration(appModel.getAppArtifact()).configMap();
+        Properties sysProps = new Properties();
+        sysProps.putAll(configMap);
+        try (CuratedApplication curatedApplication = QuarkusBootstrap.builder()
+                .setBaseClassLoader(getClass().getClassLoader())
+                .setExistingModel(appModel)
+                .setTargetDirectory(getProject().getBuildDir().toPath())
+                .setBaseName(extension().finalName())
+                .setBuildSystemProperties(sysProps)
+                .setAppArtifact(appModel.getAppArtifact())
+                .setLocalProjectDiscovery(false)
+                .setIsolateDeployment(true)
+                .build().bootstrap()) {
+            AtomicReference<List<String>> tooMany = new AtomicReference<>();
+            AugmentAction action = curatedApplication.createAugmentor();
+            action.performCustomBuild(DeployCommandDeclarationHandler.class.getName(), new Consumer<List<String>>() {
+                @Override
+                public void accept(List<String> strings) {
+                    tooMany.set(strings);
+                }
+            }, DeployCommandDeclarationResultBuildItem.class.getName());
+            String target = System.getProperty("quarkus.deploy.target");
```

Comment:
```suggestion
            String target = systemProperties.get("quarkus.deploy.target");
```
The property seems to get ignored when using `System.getProperty`. Using `systemProperties.get` seems to work better.